test/v200/utils/api/test_utils.go

```diff
@@ -0,0 +1,173 @@
+package api
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"testing"
+
+	common "github.com/devfile/api/v2/test/v200/utils/common"
+	"github.com/santhosh-tekuri/jsonschema"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// numDevfiles : the number of devfiles to create for each test
+	numDevfiles = 5
+
+	schemaFileName = "../../../schemas/latest/ide-targeted/devfile.json"
+)
+
+var schemas = make(map[string]SchemaFile)
+
+type SchemaFile struct {
+	Schema *jsonschema.Schema
+}
+
+type DevfileValidator struct{}
+
+// WriteAndVerify implements Saved.DevfileValidator interface.
+// writes to disk and validates the specified devfile
+func (devfileValidator DevfileValidator) WriteAndValidate(devfile *common.TestDevfile) error {
+	err := writeDevfile(devfile)
+	if err != nil {
+		common.LogErrorMessage(fmt.Sprintf("Error writing file : %s : %v", devfile.FileName, err))
+	} else {
+		err = validateDevfile(devfile)
+		if err != nil {
+			common.LogErrorMessage(fmt.Sprintf("Error vaidating file : %s : %v", devfile.FileName, err))
+		}
+	}
+	return err
+}
+
+// CheckWithSchema checks the validity of aa devfile against the schema.
+func (schemaFile *SchemaFile) CheckWithSchema(devfile string, expectedMessage string) error {
+
+	// Read the created yaml file, ready for converison to json
+	devfileData, err := ioutil.ReadFile(devfile)
+	if err != nil {
+		common.LogErrorMessage(fmt.Sprintf("  FAIL: schema : unable to read %s: %v", devfile, err))
+		return err
+	}
+
+	// Convert the yaml file to json
+	devfileDataAsJSON, err := yaml.YAMLToJSON(devfileData)
+	if err != nil {
+		common.LogErrorMessage(fmt.Sprintf("  FAIL : %s : schema : failed to convert to json : %v", devfile, err))
```

Comment:
simplify the message
```suggestion
		common.LogErrorMessage(fmt.Sprintf("  FAIL : schema : failed to convert %s to json : %v", devfile, err))
```

Reply:
I prefer my format.

Reply:
fine. it's up to you